src/BatchReader/Service.ts

```diff
@@ -32,6 +37,16 @@ export class Service {
   }
 
   private readNextDirectory = async () => {
-    this.messaging.publish(Exchange.BatchReaderReadNextDirectoryRequest, '')
+    const logError = (logger: Pino.Logger) => (error: any) =>
```

Comment:
try/catch and await might look a bit nicer for this change

Reply:
Sure! Let's go with that :)